### PR TITLE
Frontegg Unified Chart

### PR DESCRIPTION
**Pull request by mqxter**

### CODE REVIEW

File: charts/unified-chart/templates/high-priority/hp-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                  key: {{ $value }}
+          {{- end }}
+          {{- end }}
+          envFrom:
+            - configMapRef:
+                name: {{ include "fuc.name" . }}-configmap
+          resources:
+            {{- toYaml .Values.web.resources | nindent 12 }}
+          args:
+            - "/bin/bash"
```
Comment (guy-frontegg) [inline charts/unified-chart/templates/high-priority/hp-deployment.yaml:95]: I think we need to have override here

File: charts/unified-chart/templates/high-priority/hp-hpa.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    {{- if .Values.argoRollouts }}
+    apiVersion : argoproj.io/v1alpha1
+    kind : Rollout
+    name : {{ include "fuc.name" . }}-rollout-hp
+    {{- else }}
+    apiVersion : apps/v1
+    kind : Deployment
+    name : {{ include "fuc.name" . }}-deployment-hp
+    {{- end }}
+  minReplicas: {{ (index .Values.highPriorityMinReplicasOverride .Values.componentsCollectionIdentifier).minReplicas | default .Values.highPriority.autoscaling.minReplicas }}
```
Comment (guy-frontegg) [inline charts/unified-chart/templates/high-priority/hp-hpa.yaml:21]: we don't need it in the mew way of multiple sources

File: charts/unified-chart/templates/high-priority/hp-rollout.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+spec:
+  workloadRef: # Reference an existing Deployment using workloadRef field
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "fuc.name" . }}-deployment-hp
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      {{- include "fuc.hp.selectorLabels" . | nindent 6 }}
+  strategy:
```
Comment (guy-frontegg) [inline charts/unified-chart/templates/high-priority/hp-rollout.yaml:17]: in the future we'll need to override this, but leave it for now

File: charts/unified-chart/templates/web/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
+---
+apiVersion: apps/v1
```
Comment (guy-frontegg) [inline charts/unified-chart/templates/web/deployment.yaml:2]: same as HP

File: charts/unified-chart/templates/web/hpa.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    {{- if .Values.argoRollouts }}
+    apiVersion: argoproj.io/v1alpha1
+    kind: Rollout
+    name: {{ include "fuc.name" . }}-rollout
+    {{- else }}
+    apiVersion : apps/v1
+    kind : Deployment
+    name : {{ include "fuc.name" . }}-deployment
+    {{- end }}
+  minReplicas: {{ (index .Values.minReplicasOverride .Values.componentsCollectionIdentifier).minReplicas | default .Values.web.autoscaling.minReplicas }}
```
Comment (guy-frontegg) [inline charts/unified-chart/templates/web/hpa.yaml:20]: Same as HP

File: charts/unified-chart/templates/worker/worker-rollout.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.worker.enabled }}
+{{- if .Values.argoRollouts }}
```
Comment (guy-frontegg) [inline charts/unified-chart/templates/worker/worker-rollout.yaml:2]: don;t we have `argoRollouts.enabled`

File: charts/unified-chart/values.staging-group.godmode.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+web:
```
Comment (guy-frontegg) [inline charts/unified-chart/values.staging-group.godmode.yaml:1]: I am guessing this is for testing

File: charts/unified-chart/values.yaml [hunk 8]
```diff
@@ -0,0 +1,110 @@
+# Default values for frontegg unified chart.
+
+# required
+# serviceName: my-new-awesome-service
+# team: my-team-of-imbeciles
+
+componentsCollectionIdentifier: local
```
Comment (guy-frontegg) [inline charts/unified-chart/values.yaml:7]: do you think we should keep this?

File: charts/unified-chart/values.yaml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+minReplicasOverride:
```
Comment (guy-frontegg) [inline charts/unified-chart/values.yaml:9]: we won't need this, but i guess you keep it for legacy issues?
But we can use the new way from the start

File: charts/unified-chart/templates/_helpers.tpl [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+{{- if .Values.ingress.hostnameOverride -}}
+{{- .Values.ingress.hostnameOverride | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "api-%s.dev.frontegg.com" .Release.Name | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+
+{{/* Just the serviceName */}}
+{{- define "fuc.name" -}}
+{{- required ".Values.serviceName must be set" .Values.serviceName }}-v2
```
[the remark this answers is not in the document]
Comment (mqxter) [reply charts/unified-chart/templates/_helpers.tpl:12]: done

File: charts/unified-chart/templates/_helpers.tpl [hunk 11]
```diff
@@ -0,0 +1,102 @@
+{{/* Hostname is used in services who render ingress */}}
+{{- define "fuc.hostname" -}}
+{{- if .Values.ingress.hostnameOverride -}}
+{{- .Values.ingress.hostnameOverride | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "api-%s.dev.frontegg.com" .Release.Name | trimSuffix "-" -}}
```
[the remark this answers is not in the document]
Comment (mqxter) [reply charts/unified-chart/templates/_helpers.tpl:6]: done

File: charts/unified-chart/templates/external-secret.yaml [hunk 12]
```diff
@@ -0,0 +1,34 @@
+{{- if .Values.externalSecret }}
```
[the remark this answers is not in the document]
Comment (mqxter) [reply charts/unified-chart/templates/external-secret.yaml]: done

File: charts/unified-chart/templates/external-secret.yaml [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+metadata:
+  name: {{ include "external-secret-unique-name" . }}
+  annotations:
+    {{- with .Values.externalSecret.annotations }}
+    {{- . | toYaml | nindent 4 }}
+    {{- end }}
+  labels:
+    {{- include "fuc.labels" . | nindent 4 }}
+spec:
+  refreshInterval: 1m
```
[the remark this answers is not in the document]
Comment (mqxter) [reply charts/unified-chart/templates/external-secret.yaml:14]: done

File: charts/unified-chart/templates/high-priority/hp-deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+        {{- with .Values.web.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      terminationGracePeriodSeconds: {{ .Values.highPriority.terminationGracePeriodSeconds | default 30 }}
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- if .Values.externalSecret }}
```
[the remark this answers is not in the document]
Comment (mqxter) [reply charts/unified-chart/templates/high-priority/hp-deployment.yaml]: done

File: charts/unified-chart/templates/web/hpa.yaml (hunk 5 above)
Comment (mqxter) [reply charts/unified-chart/templates/web/hpa.yaml:20]: need more info

File: charts/unified-chart/templates/worker/worker-rollout.yaml (hunk 6 above)
Comment (mqxter) [reply charts/unified-chart/templates/worker/worker-rollout.yaml:2]: we do. its true by default

File: charts/unified-chart/values.staging-group.godmode.yaml (hunk 7 above)
Comment (mqxter) [reply charts/unified-chart/values.staging-group.godmode.yaml:1]: this is for testing godmode

**Pull request merged**